rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistStorageProvider.java

```diff
@@ -0,0 +1,8 @@
+package co.rsk.peg.whitelist;
+
+public interface WhitelistStorageProvider {
+
+    void saveLockWhitelist();
+
+    LockWhitelist getLockWhitelist();
+}
```

Comment:
We are also gonna need a new method `void save();`

Reply:
actually, the `saveLockWhitelist()` method shouldn't be part of the interface.

All the [specific `save...()` methods are private](https://github.com/rsksmart/rskj/blob/federation-support-refactor-integration/rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java#L299), and just part of the implementation.
The only one that is also part of the interface is the [`save()`](https://github.com/rsksmart/rskj/blob/federation-support-refactor-integration/rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProvider.java#L40) one, since is the one that would be [called from the supports](https://github.com/rsksmart/rskj/blob/federation-support-refactor-integration/rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java#L775)

Reply:
@julia-zack really good point.  Specific save methods should be private and only be called by the `save` method.

Reply:
Good catch

Reply:
Hey, guys! @julia-zack @nathanieliov @marcos-iov 
I was looking into BridgeStorageProvider, and I noticed that `void save()` method involves other processes apart from whitelisting. 
Should we really expose this `save()` method in `WhitelistStorageProvider`?

<img width="579" alt="image" src="https://github.com/rsksmart/rskj/assets/160710714/6d314fc3-d3c8-4a44-9a8e-1ddea13b1de4">

Reply:
I don't think I follow the question.
The `saveLockWhiteList()` method should be removed from the `BridgeStorageProvider`, since we now have specific providers. In this situation, the `WhiteListStorageProvider`.

As I said, the `WhiteListStorageProvider` interface should just have the `save()` method. The `WhiteListStorageProviderImpl` will override that method, and call the private saving methods.

See the links i provided above, showing how this was solved in the FederationSupport refactor.

Reply:
Done!